Kernel/Storage/AHCIController.cpp

```diff
@@ -124,9 +125,17 @@ AHCI::HBADefinedCapabilities AHCIController::capabilities() const
     };
 }
 
-NonnullOwnPtr<Memory::Region> AHCIController::default_hba_region() const
+NonnullOwnPtr<Memory::Region> AHCIController::default_hba_region()
 {
-    auto region = MM.allocate_kernel_region(PhysicalAddress(PCI::get_BAR5(pci_address())).page_base(), Memory::page_round_up(sizeof(AHCI::HBA)), "AHCI HBA", Memory::Region::Access::ReadWrite);
+    // The HBA address is not guaranteed to be page aligned. However, we must provide MM a page aligned physical address.
+    // We work around this by giving it a page aligned physical address, but adding the difference between the unaligned physical address and the aligned physical address to the virtual memory pointer returned by MM.
+    auto unaligned_bar5 = PhysicalAddress(PCI::get_BAR5(pci_address()));
```

Comment:
Can't we technically stop the controller, relocate the HBA BAR5 address and continue operation? It seems more wise to not deal with unaligned addresses, as it might have bad impact on performance.